File: app/src/main/java/com/main/climbingdiary/activities/SettingsActivity.java
package com.main.climbingdiary.activities;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.main.climbingdiary.common.AppPermissions;
import com.main.climbingdiary.fragments.SettingsFragment;

import java.util.Objects;

public class SettingsActivity extends AppCompatActivity {

    @SuppressLint("StaticFieldLeak")
    public static AppCompatActivity mainActivity;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SettingsActivity.mainActivity = this;
        getFragmentManager().beginTransaction().replace(android.R.id.content, new SettingsFragment()).commit();
        Objects.requireNonNull(getSupportActionBar()).setDisplayShowHomeEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        AppPermissions.checkPermissions();
    }

    public static AppCompatActivity getSettingsActivity(){return SettingsActivity.mainActivity;}

    @Override
    public void onBackPressed() {
        finish();
    }

    @Override
    public boolean onSupportNavigateUp() {
        finish();
        return true;
    }
}

File: app/src/main/java/com/main/climbingdiary/activities/SettingsActivity.kt
package com.main.climbingdiary.activities

//noinspection SuspiciousImport
import android.R
import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.main.climbingdiary.common.AppPermissions
import com.main.climbingdiary.fragments.SettingsFragment

class SettingsActivity : AppCompatActivity() {

    companion object {
        lateinit var mInstance: SettingsActivity

        @JvmStatic
        fun getSettingsActivity(): Context {
            return mInstance.applicationContext
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mInstance = this
        supportFragmentManager.beginTransaction().replace(R.id.content, SettingsFragment()).commit()
        supportActionBar?.setDisplayShowHomeEnabled(true)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        Thread {
            this@SettingsActivity.runOnUiThread {
                AppPermissions.checkPermissions(this)
            }
        }.start()
    }

    @Deprecated("Deprecated in Java")
    override fun onBackPressed() {
        super.onBackPressed()
        finish()
    }

    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}